plugins/ffmpeg_plugin/kahawai.c

```diff
@@ -0,0 +1,606 @@
+/*
+ * Kahawai raw video demuxer
+ * Copyright (c) 2022 Intel
+ *
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <st_convert_api.h>
+#include <st_pipeline_api.h>
+
+#include "libavformat/avformat.h"
+#include "libavformat/internal.h"
+#include "libavutil/imgutils.h"
+#include "libavutil/opt.h"
+#include "libavutil/parseutils.h"
+#include "libavutil/pixdesc.h"
+
+#define KAHAWAI_FRAME_BUFFER_COUNT 8
+#define KAHAWAI_FPS_SUPPORTED 5
+
+const static unsigned int fps_table[KAHAWAI_FPS_SUPPORTED][3] = {
+    {ST_FPS_P59_94, 5994 - 100, 5994 + 100},    {ST_FPS_P50, 5000 - 100, 5000 + 100},
```

Comment:
This really hard to understand, can you help to change to struct?
struct fps_decs {
  enum ST_FPS fps;
  unsigned int a;
  unsigned int b;
};

Reply:
Modified.